src/utilities.js

```diff
@@ -69,68 +53,79 @@ const filterObjectsFromSave = function (save, list) {
 module.exports.filterObjectsFromSave = filterObjectsFromSave;
 
 /**
- *
- * @param {string} type type of validation that is requested
- * @param {float|number} min minimum value a "number" type that can take
- * @param {float|number} max maximum value a "number" type that can take
- * @param selected item value to be checked
- * @param previous default value if item is undefined&NaN
- * @returns {string|*} valid result
+ * Returns if object is number or not
+ * @param {string}n String to be checked
+ * @returns {boolean}
  */
-const getValidData = function (type, min, max, selected, previous) {
-    switch (type) {
-        case "number":
-            return validateNumber(parseFloat(selected), min, max, previous);
-        case "text":
-            if (selected) {
-                return selected.toString();
-            } else if (previous) {
-                return previous.toString();
-            } else {
-                return "";
-            }
-        default:
-            return selected;
-    }
+const isNumeric = function (n) {
+    return !isNaN(parseFloat(n)) && isFinite(n);
 };
 
-module.exports.getValidData = getValidData;
+module.exports.isNumeric = isNumeric;
 
 /**
  * Returns a valid number within constrains
  * @param selected value to be checked
  * @param min minimum value allowed for numbers
  * @param max maximum value allowed for numbers
- * @param previous last valid entry
  * @returns {number|float} valid number
  */
-const validateNumber = function (selected, min, max, previous) {
-    if (isNaN(selected)) {
-        if (previous !== undefined) {
-            if (!isNaN(parseFloat(previous))) {
-                return validateNumber(parseFloat(previous), min, max, undefined);
-            } else {
-                return 0;
-            }
+const clampNumber = function (selected, min, max) {
+    if (selected < min) {
+        return min;
+    } else if (selected > max) {
+        return max;
+    } else {
+        return selected;
+    }
```

Comment:
eslint may warn the else-return

```js
if (selected < min) {
   return min;
}
if (selected > max) {
    return max;
}
return selected;
```

- no "else if" chain, it's ok for this simple case and easy to move the code block.
  - change max-check block first, if it's used more than min-check. (very small optimization)
    it has extra format task if modify "else if" chain.
  - but result may change by which min or max first, when invalid parameter given like "min > max"
- (just info) single return is better sometime, if possible. I explain merit of lodash clamp approach

```js
if (num < min) {
    num = min;
}
if (num > max) {
    num = max;
}
// can be break-point for debug, all cases pass here.
// print debug too, if three return statements, then logging points also three.
return num;
```

no need to be too strict in this simple function.

Reply:
it was up to the option: https://eslint.org/docs/rules/no-else-return

Reply:
Fixed in 8902cef

Reply:
Lodash approach doesn't work if any of the min, max values are undefined which is a valid use case.

Reply:
this is just show merit of single return.
undefined check is not responsibility of this function, callee should check
and in use case max or min undefine, should use Math.min/max, clamp for both check at once.

Reply:
no need "selected" variable name, as general utility function.
not selected component specific code now.